client/app/utils/itemClasses.js

```diff
@@ -0,0 +1,177 @@
+         const node = this.get('node');
+         let self = this;
+         loadAll(node, 'contributors', contributors).then(function (){
+             self.get('viewContent.authors').setValue(contributors);
+         });
+     },
+     setCommonNodeContent(node){
+         this.set('node', node);
+         this.get('viewContent.description').setValue(node.get('description'));
+         this.get('viewContent.tags').setValue(node.get('tags'));
+         this.setAuthors();
+     },
+     resetContent(){
+         this.set('viewContent', Ember.Object.create({
+            title: ViewData.create(),
+            description: ViewData.create(),
+            tags: ViewData.create(),
+            authors: ViewData.create(),
+            sourceLink: ViewData.create(),
+            wiki: ViewData.create({visible:false}),
+            file: ViewData.create({visible:false})
+        }));
+     },
+     init (){
+         this.resetContent();
+         this.get('viewContent.title').setValue(this.get('item.title'));
+     }
+ });
+
+ /*
+  *  Builds data points for website type item
+  */
+ const Website = Item.extend({
+     init(){
+         this._super();
+         this.get('viewContent.description').setValue(this.get('item.metadata'));
+         this.get('viewContent.sourceLink').setValue(this.get('item.source_id'));
+         this.get('viewContent.tags').hide();
+         this.get('viewContent.authors').hide();
+     }
+ });
+
+ /*
+  *  Builds data points for project ('node') type item
+  */
+ const Project = Item.extend({
+     setWiki (){
+         let self = this;
+         let node = this.get('node');
+         let wikis = node.get('wikis');
+         self.get('viewContent.wiki').show();
+         if(wikis){
+             wikis.then(function(result){
+                 if(result.objectAt(0)){
+                     let url = result.objectAt(0).get('links.download');
+                     let headers = {};
+                     let authType = config['ember-simple-auth'].authorizer;
+                     self.get('session').authorize(authType, (headerName, content) => {
+                         headers[headerName] = content;
+                     });
+
+                     Ember.$.ajax({
+                         method: 'GET',
+                         headers,
+                         url
+                     }).done(data => {
+                         self.get('viewContent.wiki').setValue(data);
+                     });
+                 } else {
+                     self.get('viewContent.wiki').setValue('This project does not have wikis.');
+                 }
+             }.bind(this));
+         } else {
+             self.get('viewContent.wiki').setValue('Could not find wiki for this project.');
+         }
+     },
+     init(){
+         this._super();
+         let self = this;
+         this.get('store').findRecord('node', this.get('item.source_id')).then(function(node){
+             self.setCommonNodeContent(node);
```

Comment:
If you use ES6 arrow syntax here, you can use `this` without having to assign it to `self`